x-pack/packages/kbn-entities-schema/src/schema/entity_definition.ts

```diff
@@ -32,22 +31,18 @@ export const entityDefinitionSchema = z.object({
   metrics: z.optional(z.array(keyMetricSchema)),
   staticFields: z.optional(z.record(z.string(), z.string())),
   managed: z.optional(z.boolean()).default(false),
-  history: z.object({
```

Comment:
Q: Since we're dropping this (previously required) field. 

What would occur in the event of a rollback (if we have to revert to the current version after deploying the version including these changes, and having created a few entities)?

Will the newly formatted entities be considered corrupt by Kibana?
If so, can Kibana self-recover? Or will users suffer from data loss?

Reply:
There's a few difference regarding how the fields are aggregated in the new version, but overall entities keep the same schema (minus one field that's removed in the new version), so rollback/rollforward would be seamless as far as stored documents goes.

The entity definitions installed in the new version would be somewhat broken in the rolled back environment (few apis won't be able to interact with those). I didn't implement a good story for this since 1. this is tech preview 2. we don't have released product built on these yet.

Where can I read about our rollback strategy ?

Reply:
@kobelb, @rudolf, with those premises (unreleased tech preview), is it OK to make these changes in our SO schema?

Reply:
At the end of the day making breaking changes to internal APIs or disrupting the UI during a rollback is a product decision. The team responsible for it is best able to make this tradeoff. The important thing is that we do weigh up such customer impact instead of just shipping something. It sounds like the team has done this 👍🏻